python_fundamentals-master/08_file_io/08_01_words_analysis.py

```diff
@@ -7,3 +7,20 @@
 
 
 '''
+with open("words.txt" , "r") as f:
+    word_min = "hello"
```

Comment:
why is `word_min` hardcoded to be `hello` ?
I think you were trying to find a large word to start with? In that case I would recommend you use `float('inf')` which is pythons positive infinity.  There is also a `float('-inf')` for negative infinity. These are commonly used for this purpose, setting limits.

![image](https://user-images.githubusercontent.com/6275366/122549115-c3266800-d04f-11eb-8ff6-b8accfba5a22.png)

Reply:
Yes I was looking for a word longer than one to have a starting point, and couldn't think of a better option! ('inf') is great!

I tried to use float('inf') just now, but got this error:
    if len(word) <= word_min:
TypeError: '<=' not supported between instances of 'int' and 'str'

I made it into str('inf') - it did do the trick, so I assume is works the same :)

Reply:
no they are not at all the same.  You are doing string comparison instead of integer comparison.
it will compare the ASCII values of the two stings and use that to evaluate <, > etc.

You got the error because you were trying to compare an integer with a string.  You should convert the other string (`word_min`) into an integer.  You probably need `len(word_min)` so you are comparing the lengths of the words.